_posts/2024-03-25-whats-cooking.md

```diff
@@ -0,0 +1,224 @@
+
+- **Secret Manager:** Helps to protect access to applications, services, and IT resources without the upfront cost and complexity of managing a full infrastructure.
+
+- **DynamoDB:** A fast and flexible NoSQL database service for all applications that need consistent, single-digit millisecond latency at any scale.
+
+- **API Gateway:** Offers a fully managed service that makes it easy for developers to create, publish, maintain, monitor, and secure APIs at any scale.
+
+- **API Gateway API Key:** Utilized for controlling and monitoring API usage and access, enhancing security and usage metrics.
+
+### CI/CD Pipeline
+
+- **GitHub Workflows:** Automates your software development workflows.
+
+- **Terraform:** An infrastructure as code software tool that allows you to build, change, and version infrastructure safely and efficiently.
+
+### AI Models
+
+- **OpenAI Models:** Provides advanced AI models like GPT (Generative Pre-trained Transformer) for natural language understanding, generation, and conversational AI capabilities.
+
+- **AWS AI Model Bedrock Titan:** Bedrock Titan AI is a suite of large language models offered by Amazon for building generative AI applications.
+
+- **Cohere Command:** Cohere Command is a powerful language model designed for enterprises, excelling at following instructions and completing complex tasks.
+
+- **Meta Llama2:** Meta Llama2 is a next-generation large language model known for its commercial-grade capabilities and open-source availability.
+
+# Application Architecture (DRAFT - Febe)
+
+In our project, we implemented CI/CD to streamline the deployment process of our backend.
+
+This was achieved by leveraging various tools offered by different providers such as GitHub and Amazon Web Services (AWS).
+
+Initially, we began by writing workflow files to automate the execution of tests with each push to the development branch on GitHub.
+
+Over time, this progressed into developing workflow files aimed at building and deploying our backend onto App Runner, thereby enabling seamless access to our backend without the necessity of local execution.
+
+Within our GitHub workflow files, we designed jobs dedicated to constructing Docker images using Maven, facilitating the subsequent pushing of the latest backend image to Amazon Elastic Container Registry (ECR).
+
+Following the image push to ECR, we employed Terraform to deploy this image onto App Runner.
+
+To streamline the orchestration of these jobs, which were authored in separate files for organizational clarity, we utilized a main workflow file to dictate the sequence in which the jobs were to be executed.
+
+In our pursuit of enhancing backend security on App Runner, we opted to integrate an API Gateway.
+
+This gateway serves as a protective layer, mandating that all frontend requests transit through it and ensuring access to the backend only if the appropriate key is present in the request headers.
+
```

Comment:
Voeg hier ook zeker een diagram toe van de infrastructuur die jullie hebben opgezet.
draw.io is een goeie tool om dit te maken.